UI.Web/ClientApp/src/app/components/tic-tac-toe/board/board.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+    selector: 'board',
+    templateUrl: './board.component.html',
+    styles: [
+        `main {display: grid;
+    grid-template-columns: 200px 200px 200px;}`
+    ]
+})
+
+export class BoardComponent implements OnInit {
+    squares: any[];
+    xIsNext: boolean;
+    winner: string;
+
+    constructor() {}
+
+    ngOnInit() {
+        this.newGame();
+    }
+
+    newGame() {
+        this.squares = Array(9).fill(null);
+        this.winner = null;
+        this.xIsNext = true;
+    }
+
+    get player() {
+        return this.xIsNext ? 'X' : 'O';
+    }
+
+    // method serves as event handler when click a button
+    // check index in the array they've clicked on
+    // if already clicked do nothing
+    // if null, splice in the index of the square with the player (x or o)
+    makeMove(index: number) {
+        if (!this.squares[index]) {
+            this.squares.splice(index, 1, this.player)
+            this.xIsNext = !this.xIsNext;
+        }
+
+        this.winner = this.calculateWinner();
+    }
+
+    calculateWinner() {
+        const lines = [
```

Comment:
This is way more clever of a way to check for winners than what we did initially in the previous one.